front50-api/src/main/java/com/netflix/spinnaker/front50/api/model/pipeline/Trigger.java

```diff
@@ -28,6 +32,76 @@ protected Map<String, Object> delegate() {
     return backing;
   }
 
+  public Boolean getEnabled() {
+    Object enabled = backing.get("enabled");
+    if (enabled instanceof Boolean) {
+      return (Boolean) enabled;
+    }
+
+    // orca's DependentPipelineExecutionListener.groovy currently (18-apr-23)
+    // has code like
+    //
+    // if (trigger.enabled)
+    //
+    // which means any non-null, non-empty string/map/list/non-zero number is
+    // truthy.  See https://groovy-lang.org/semantics.html#the-groovy-truth.
+    // It's likely worth changing orca as this is not the most obvious behavior.
+    // Whether orca changes or not, it doesn't seem right to duplicate that
+    // non-obvious behavior here.
+    //
+    // Perhaps the most surprising part of this behavior is that a string
+    // "false" is interpreted at truthy.  To that end, let's add special
+    // handling for strings so "false" is considered false, and "true" is
+    // considered true.
+    //
+    // Also note that if we stop extending ForwardingMap here and convert to a
+    // "proper" pojo, this method would likely disappear and jackson would take
+    // over, handling true/True/TRUE and false/False/FALSE and throwing an
+    // exception otherwise.
+    //
+    // Getting an objectMapper here to invoke explicitly would be nice in that
+    // we'd only have one behavior change, but code-wise it feels
+    // cumbersome.
+    //
+    // So, let's try to make the trajectory here from less restrictive to more
+    // restrictive without then becoming less restrictive in the future.  In
+    // other words, we start by considering the most triggers enabled, and
+    // reduce that over time, but once a trigger is considered disabled, it
+    // doesn't magically become enabled again.  So, let's consider at least as
+    // many triggers enabled as jackson would.
+    if (enabled instanceof String) {
+      // This allows not only true/True/TRUE but also some values that groovy
+      // considers truthy, e.g. tRue, tRUe, etc.
+      return Boolean.valueOf(((String) enabled).toLowerCase());
+    }
+
+    if (enabled != null) {
+      // It's tempting to warn if enabled is some other type, as it's likely a
+      // pipeline bug.  There's not enough context here to help find the
+      // offending pipeline though, so let's leave it as debug.
+      log.debug(
```

Comment:
Should be a WARN vs. debug?  Possibly including the name of the pipeline or some other details be ideal as well.

Reply:
Right...the comment is trying to explain this.  I can go back to warn, but there isn't really any other context here.  If we're lucky the log messages would have e.g. X-SPINNAKER-REQUEST-ID.

Reply:
yeah I THOUGHT there was context in the backing object :( IN THEORY X-Spinnaker-Applicaiton or similar headers?

Reply:
https://github.com/spinnaker/front50/pull/1251/commits/f947b03ef566b04ea64030c85d4eeb2313b501f3